src/plugin.py

```diff
@@ -175,41 +181,45 @@ async def _handle_confirmation_result(self):
         self._use_paranoid_login = True
 
         username_or_display = self._view.retrieve_data_paranoid_username(credentials)
         if isinstance(username_or_display, NextStep):
             return username_or_display
+
         username = cast(str, username_or_display)
         key_or_error = await self._model.retrieve_rsa_key(username)
-        if isinstance(key_or_error, SteamPublicKey):
-            key_data = cast(SteamPublicKey, key_or_error)
-            return self._view.paranoid_username_success(username, key_data.rsa_public_key, key_data.timestamp)
-        else:
+        if isinstance(key_or_error, ModelAuthError):
             return self._view.paranoid_username_failed(cast(ModelAuthError, key_or_error))
 
-    async def _handle_manual_deciphering_result(self, credentials: Dict[str, str])  -> Union[Authentication, NextStep]:
+        key_data = cast(SteamPublicKey, key_or_error)
+        return self._view.paranoid_username_success(username, key_data.rsa_public_key, key_data.timestamp)
+
+    async def _handle_manual_enciphering_result(self, credentials: Dict[str, str]) -> Union[Authentication, NextStep]:
         fallback_or_data = self._view.retrieve_data_paranoid_pt2(credentials)
-        if (isinstance(fallback_or_data, NextStep)):
+        if isinstance(fallback_or_data, NextStep):
             return fallback_or_data
-        (username, enciphered_password, timestamp) = cast(Tuple[str, bytes, int], fallback_or_data)
+
+        username, enciphered_password, timestamp = cast(Tuple[str, bytes, int], fallback_or_data)
         return await self._do_common_credential_login(username, enciphered_password, timestamp, True)
 
-    async def _do_common_credential_login(self, username : str, enciphered_password : bytes, timestamp: int, is_paranoid_user_result: bool) -> Union[Authentication, NextStep]: #testing, needs revert. 
-        two_factor_data_or_error = await self._model.login_with_credentials(username, enciphered_password, timestamp) #testing, needs revert
+    async def _do_common_credential_login(self, username: str, enciphered_password: bytes, timestamp: int, is_paranoid_user_result: bool) -> Union[Authentication, NextStep]:  # testing, needs revert.
+        two_factor_data_or_error = await self._model.login_with_credentials(username, enciphered_password, timestamp)  # testing, needs revert
         if isinstance(two_factor_data_or_error, ModelAuthCredentialData):
             self._two_factor_info = cast(ModelAuthCredentialData, two_factor_data_or_error)
             self._unauthed_steam_id = self._two_factor_info.steam_id
 
             auth_methods = self._two_factor_info.allowed_authentication_methods
-            if not auth_methods or not auth_methods[0] or auth_methods[0].confirmation_type == EAuthSessionGuardType.k_EAuthSessionGuardType_Unknown:
+            if len(auth_methods) == 0 or auth_methods[0].confirmation_type == EAuthSessionGuardType.k_EAuthSessionGuardType_Unknown:
```

Comment:
iirc len == 0 is handled as Falsy and so is None. 
I believe i meant `not auth_methods or auth_methods[0] is None`
In theory these should never be None because that'd mean either Steam or betterproto gave us `null` (which is represented in python as None, ofc), and i highly doubt that would happen... *but* i've grown accoustomed to using the elvis operator when i can't be sure the source cannot be null and this is the closest equivalent in python.

Reply:
oops, mistake on my end. the comparison must be `!= 0`. good catch